torchinductor/lowering.py

```diff
@@ -981,15 +1009,54 @@ def inner(
         dtype = dtype or torch.get_default_dtype()
         if len(size) == 1 and isinstance(size[0], (list, tuple, torch.Size)):
             size = tuple(size[0])
-        size = [sympy.Integer(s) for s in size]
+        size = [sympy.expand(s) for s in size]
         return _full(fill_value, device, dtype, size)
 
     return inner
 
 
-register_lowering([torch.empty, aten.empty])(tensor_constructor(0))
-register_lowering([torch.zeros, aten.zeros])(tensor_constructor(0))
-register_lowering([torch.ones, aten.ones])(tensor_constructor(1))
+empty = register_lowering([torch.empty, aten.empty])(tensor_constructor(0))
+zeros = register_lowering([torch.zeros, aten.zeros])(tensor_constructor(0))
+ones = register_lowering([torch.ones, aten.ones])(tensor_constructor(1))
+
+
+def create_tensor_like(creation_fn):
+    """
+    Shim to convert X_like(...) into X(...).  For example zeros_like() into zeros().
+    """
+
+    def _constant_like(
+        x, *, dtype=None, device=None, layout=0, pin_memory=False, memory_format=None
+    ):
+        assert not pin_memory
+        assert layout in (0, torch.strided)
+        assert memory_format in (None, torch.contiguous_format)
+        if dtype is None:
+            dtype = x.get_dtype()
+        else:
+            dtype = decode_dtype(dtype)
+        device = device or x.get_device()
+        size = list(x.get_size())
+        return creation_fn(
+            size,
```

Comment:
this won't correctly reproduce *_like behavior, because by default *_like functions preserve permutation of original tensor (e.g. `zeros_like(x)` would return channels-last if `x` was channels last, or transposed if `x` was transposed 2d tensor)

Reply:
This is true, though decomps in general get strides wrong, so this won't be the only case like this.

Currently, none of tensor constructors support layout specifiers (they all have asserts they are unset or contiguous).  Adding support for specifying layouts is not yet implemented and future work.